hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -247,6 +245,9 @@ object DefaultSource {
       Option(schema)
     }
 
+    val useMORBootstrapFF = parameters.getOrElse(MOR_BOOTSTRAP_FILE_READER.key,
+      MOR_BOOTSTRAP_FILE_READER.defaultValue).toBoolean && (globPaths == null || globPaths.isEmpty)
```

Comment:
It can be fixed, but I attempted it for a bit and it seemed like it would take .5-1 days. Since that uses in memory file index, I assume that glob paths only happens for small tables that wouldn't benefit much from this improvement anyways?

Reply:
Got it.  Then let's leave this as a follow-up.  The new file format should support this too for feature completeness.